src/Microsoft.Azure.WebJobs/ConnectionProviderAttribute.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// For Attributes that implement <see cref="IConnectionProvider"/> this attribute can be
+    /// applied to the Attribute to specify the attribute Type used for hierarchical
+    /// declarative overrides.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ConnectionProviderAttribute : Attribute
```

Comment:
Assuming we want to mark the ConnectionProvider classes as obsolete. Is this intended as a mechanism for an internal fix that's only public so that we don't duplicate across assemblies? Or do we want to expose this to extension authors as a mechanism for hierarchical connection overrides (I'd rather avoid this)?